src/BuiltInTools/dotnet-watch/Internal/ConsoleReporter.cs

```diff
@@ -49,17 +69,36 @@ protected virtual void WriteLine(TextWriter writer, string message, ConsoleColor
             }
         }
 
+        private string GetEmojiPrefix(string message)
+        {
+            foreach (var (prefix, emoji) in PrefixEmojiAssociations)
+            {
+                if (message.StartsWith(prefix, StringComparison.Ordinal))
```

Comment:
This feels a little iffy. Is it too much work just to include the icons on individual statements? There's not too many of them, are there?

Reply:
I had originally done it that way, but it had a bunch of complications. We either needed to expand the logging API to take the emoji as an arg, or do inspection anyways to see if an emoji is already present (otherwise add the default ⌚ emoji). I found this way to be a much cleaner solution, and I don't think there should be any (major) perf implications.

Reply:
Discussed offline, updating to use an emoji arg based approach to facilitate potential localization.